### PR TITLE
JSON parser: Use array for function params (#2255)

### PR DESCRIPTION
**Pull request by gtrxAC**
Related issue: https://github.com/raysan5/raylib/issues/2255
Uses arrays with objects containing name and type instead of objects with name as key and type as value for function parameters in the JSON raylib parser.

### CODE REVIEW

File: parser/raylib_parser.c [hunk 1]
```diff
@@ -1001,14 +1001,17 @@ static void ExportParsedData(const char *fileName, int format)
...
                 else
                 {
-                    fprintf(outFile, ",\n      \"params\": {\n");
+                    fprintf(outFile, ",\n      \"params\": [\n");
                     for (int p = 0; p < funcs[i].paramCount; p++)
                     {
-                        fprintf(outFile, "        \"%s\": \"%s\"", funcs[i].paramName[p], funcs[i].paramType[p]);
+                        fprintf(outFile, "        {\n");
+                        fprintf(outFile, "          \"name\": \"%s\",\n", funcs[i].paramName[p]);
+                        fprintf(outFile, "          \"type\": \"%s\"\n", funcs[i].paramType[p]);
```
Comment (raysan5) [inline parser/raylib_parser.c:1009]: Maybe we could exchange order of those lines to follow C convention of (`type` `name`).

File: parser/raylib_api.lua [hunk 2]
```diff
@@ -2636,6 +2636,14 @@ return {
         {name = "height", type = "int"}
       }
     },
+    {
+      name = "SetWindowOpacity",
+      description = "Set window opacity [0.0f..1.0f] (only PLATFORM_DESKTOP)",
+      returnType = "void",
+      params = {
+        {name = "opacity", type = "float"}
```
Comment (raysan5) [inline parser/raylib_api.lua:2644]: Maybe we could exchange order of those lines to follow C convention of (`type` `name`).

File: parser/raylib_api.json [hunk 3]
```diff
@@ -2490,11 +2490,20 @@
...
       "description": "Initialize window and OpenGL context",
       "returnType": "void",
-      "params": {
-        "width": "int",
-        "height": "int",
-        "title": "const char *"
-      }
+      "params": [
+        {
+          "name": "width",
```
Comment (raysan5) [inline parser/raylib_api.json]: Maybe we could exchange order of those lines to follow C convention of (`type` `name`).

**Comment by raysan5** [comment]
@gtrxAC Thanks for the improvement!

**Comment by planetis-m** [comment]
So this broke my parser as expected, but the issue is the params are now flipped, instead of name, type which what FieldInfo uses by the way, they are now type, name. At least make the consistent and if possible don't follow the C syntax. For example `{"name": "name[10]", "type": "char"` is hard to parse because we need to take into account the field "name" as well to infer the type.
